core/templates/dev/head/pages/admin-page/roles-tab/admin-roles-tab.directive.ts

```diff
@@ -42,40 +43,44 @@ oppia.directive('adminRolesTab', [
       controllerAs: '$ctrl',
       controller: [function() {
         var ctrl = this;
-        ctrl.UPDATABLE_ROLES = GLOBALS.UPDATABLE_ROLES;
-        ctrl.VIEWABLE_ROLES = GLOBALS.VIEWABLE_ROLES;
-        ctrl.topicSummaries = GLOBALS.TOPIC_SUMMARIES;
-        ctrl.graphData = GLOBALS.ROLE_GRAPH_DATA;
+
         ctrl.resultRolesVisible = false;
         ctrl.result = {};
         ctrl.setStatusMessage('');
         ctrl.viewFormValues = {};
         ctrl.updateFormValues = {};
         ctrl.viewFormValues.method = 'role';
 
-        ctrl.graphDataLoaded = false;
-        // Calculating initStateId and finalStateIds for graphData
-        // Since role graph is acyclic, node with no incoming edge
-        // is initState and nodes with no outgoing edge are finalStates.
-        var hasIncomingEdge = [];
-        var hasOutgoingEdge = [];
-        for (var i = 0; i < ctrl.graphData.links.length; i++) {
-          hasIncomingEdge.push(ctrl.graphData.links[i].target);
-          hasOutgoingEdge.push(ctrl.graphData.links[i].source);
-        }
-        var finalStateIds = [];
-        for (var role in ctrl.graphData.nodes) {
-          if (ctrl.graphData.nodes.hasOwnProperty(role)) {
-            if (hasIncomingEdge.indexOf(role) === -1) {
-              ctrl.graphData.initStateId = role;
-            }
-            if (hasOutgoingEdge.indexOf(role) === -1) {
-              finalStateIds.push(role);
+        AdminDataService.getDataAsync().then(function(response) {
+          ctrl.UPDATABLE_ROLES = response.updatable_roles;
```

Comment:
I think we have missed this one, can you please initialize the values above?

Reply:
Done